### DISCUSSION TITLE
EPS8266 WIFI Connection

### DISCUSSION THREAD

**Discussion by lassekite** [micropython/micropython#9908] [General]
Hi, I have some fishy problem with WIFI/MQTT connection with a Lolin D1 mini powered with Lipo via a battery sheild. I have a switch to the battery and when I turn on the power it almost never connect but if I push reset it always does.  It's already mounted in box so it's a bit of a problem to fix a extra switch for reset now. 
Any idea?
/Lasse

**Comment by davefes** [comment]
Show us your WiFi connect code.

**Comment by peterhinch** [comment]
This sounds like an electrical problem. When you switch it on does your code actually run? If the voltage ramps up slowly it's possible that it crashes: I'd put in some kind of check that this hasn't happened like flashing an LED or outputting print statements.

A possible workround might be to put in a delay in `main.py` to let the voltage stabilise before using the WiFi (which uses more current):
```python
import time
time.sleep(4)
import my_application
```

**Reply by lassekite** [reply]
I have code that runs without connection, so yes it runs. Don't see how the Iipo rump up but I get it a try with the sleep.
This is the conc code:

def conwifi2(ssid, pw):
	import network
	from time import sleep_ms
	sts = ('IDLE','CONNECTING','WRONG_PASSWORD','NO_AP_FOUND','CONNECT_FAIL','GOT_IP')
	sta_if = network.WLAN(network.STA_IF)
	if not sta_if.isconnected():
		print('* connecting to network...', ssid)
		sta_if.active(True)
		sta_if.connect(ssid, pw)
		if not sta_if.isconnected():
			sleep_ms(1000)
	print('** network config:', sta_if.ifconfig())
	print('*** status: ', sts[sta_if.status()])
	# if sta_if.isconnected():
	if sta_if.status() == 5:
		return True
	else:		
		return False


Thank's for your replys

**Reply by davefes** [reply]
Things to try:
```py
    if sta_if.isconnected():
        sta_if.disconnect()
        print ('started in the connected state')
        print ('but now it is disconnected')
    else:
        print ('started in the disconnected state')
```
placed before your:
> if not sta_if.isconnected():

Also,, try the following at the very beginning:
```py
ap_if = network.WLAN(network.AP_IF

if ap_if.isconnected():
        ap_if.disconnect()
        print ('started in the connected state')
        print ('but now it is disconnected')
    else:
        print ('started in the disconnected state')
```

There can be many hardware issues.  All my ESP32 boards have a 1000uF low-esr capacitor on both the 5V and 3V3 board connections to help the WiFi get through power-up surges.

**Comment by peterhinch** [comment]
To back up my previous comment, the reset pin on a microcontroller forces it to behave exactly as it does on power up. If a program runs after pressing the reset button that confirms a working application. If it doesn't work on power up, you can be highly confident that there is an electrical issue. Further, an ESPx chip draws a high current when it connects to WiFi. So I would suggest that, by connecting immediately, you are drawing a high current at a time when capacitors are still charging and the power supply is not yet stable. This is why I suggested a delay.

Power supply stability could readily be proved or disproved with an oscilloscope.

**Reply by lassekite** [reply]
Fantastic it seems to work with the delay, thank for your help
/LA
